cmd/telemetrygen/internal/logs/logs.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen/internal/common"
+)
+
+type exporter interface {
+	export(plog.Logs) error
+}
+
+type gRPCClientExporter struct {
+	client plogotlp.GRPCClient
+}
+
+func (e *gRPCClientExporter) export(logs plog.Logs) error {
+	req := plogotlp.NewExportRequestFromLogs(logs)
+	if _, err := e.client.Export(context.Background(), req); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Start starts the log telemetry generator
+func Start(cfg *Config) error {
+	logger, err := common.CreateLogger()
+	if err != nil {
+		return err
+	}
+
+	if cfg.UseHTTP {
+		return fmt.Errorf("http is now not supported in log mode of telemetrygen")
+	}
+
+	if !cfg.Insecure {
+		return fmt.Errorf("the log mode of telemetrygen only supports insecure grpc")
```

Comment:
Open another issue for adding support for this?
```suggestion
		return fmt.Errorf("'telemetrygen logs' only supports insecure gRPC")
```

Reply:
here is the issue #18868